File: codewars/6kyu/write-number-in-expanded-form/dduck-write-number-in-expanded-form.js
// write-number-in-expanded-form
// https://www.codewars.com/kata/5842df8ccbd22792a4000245/

const expandedForm = (num) => String(num).split('')
  .reduceRight((a, c, i, arr) => {
    if (c === '0') {
      return a;
    }
    let result = `${c}${'0'.repeat(arr.length - 1 - i)}`;
    if (a !== '0') {
      result = `${result} + ${a}`;
    }
    return result;
  });

console.log(expandedForm(12), '10 + 2');
console.log(expandedForm(42), '40 + 2');
console.log(expandedForm(70304), '70000 + 300 + 4');
console.log(expandedForm(9000000));

File: codewars/6kyu/write-number-in-expanded-form/dduck-write-number-in-expanded-form.py
# write-number-in-expanded-form
# https://www.codewars.com/kata/5842df8ccbd22792a4000245/

from unittest import TestCase


def expanded_form(num):
    result = []
    for i, v in enumerate(list(str(num))[::-1]):
        if v != "0":
            result.append(v + "0" * i)
    return " + ".join(result[::-1])


TestCase().assertEqual(expanded_form(12), '10 + 2')
TestCase().assertEqual(expanded_form(42), '40 + 2')
TestCase().assertEqual(expanded_form(70304), '70000 + 300 + 4')
